features/step_definitions/registry.rb

```diff
@@ -397,3 +397,45 @@
   cb_name ||= :generation_number
   cb[cb_name] = deployment(name).generation_number(user: user, cached: false)
 end
+
+Given /^docker config for default image registry is stored to the#{OPT_SYM} clipboard$/ do |cb_name|
+
+  step %Q/I enable image-registry default route/
+  step %Q/default image registry route is stored in the :integrated_reg_ip clipboard/
+  step %Q/a 5 characters random string of type :dns is stored into the :dockercfg_name clipboard/
+  cb[cb_name] ="/tmp/#{cb.dockercfg_name}"
```

Comment:
@xiuwang thanks for the update, but this step will fail if `cb_name` is not specified which will become `nil`.  You should add this a line like `cb_name ||= :dockercfg_name` Here's a good article if that explains what `||=` does http://www.rubyinside.com/what-rubys-double-pipe-or-equals-really-does-5488.html

Reply:
@pruan-rht Thanks, Peter~ Have updated all comments